api/core/v1alpha1/clustercatalog_types.go

```diff
@@ -85,153 +110,225 @@ type ClusterCatalogSpec struct {
 	//    image:
 	//      ref: quay.io/operatorhubio/catalog:latest
 	//
-	// For more information on FBC, see https://olm.operatorframework.io/docs/reference/file-based-catalogs/#docs
+	// +kubebuilder:validation:Required
 	Source CatalogSource `json:"source"`
 
-	// priority is an optional field that allows the user to define a priority for a ClusterCatalog.
+	// priority allows the user to define a priority for a ClusterCatalog.
+	// priority is optional.
+	//
 	// A ClusterCatalog's priority is used by clients as a tie-breaker between ClusterCatalogs that meet the client's requirements.
+	// It is up to clients to decide how to handle scenarios where multiple ClusterCatalogs with the same priority meet
 	// For example, in the case where multiple ClusterCatalogs provide the same bundle.
 	// A higher number means higher priority. Negative numbers are also accepted.
+	//
 	// When omitted, the default priority is 0.
+	//
 	// +kubebuilder:default:=0
 	// +optional
 	Priority int32 `json:"priority"`
 
-	// Availability is an optional field that allows users to define whether the ClusterCatalog is utilized by the operator-controller.
+	// availabilityMode allows users to define the of the ClusterCatalog.
```

Comment:
Missing some words here
```suggestion
	// availabilityMode allows users to define the how the ClusterCatalog is made available to clients on the cluster.
```